src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -82,6 +81,19 @@ namespace winrt::TerminalApp::implementation
                 _dismissPalette();
             }
         });
+
+        // Focusing the ListView when the Command Palette control is set to Visible
+        // for the first time fails because the ListView hasn't finished loading by
+        // the time Focus is called. Luckily, We can listen to SizeChanged to know
+        // when the ListView has been measured out and is ready, and we'll immediately
+        // revoke the handler because we only needed to handle it once on initialization.
+        _sizeChangedRevoker = _filteredActionsView().SizeChanged(winrt::auto_revoke, [this](auto /*s*/, auto /*e*/) {
```

Comment:
Have you tried `LayoutUpdated`? It might be better for a oneshot event for when the listview is "ready" :smile:

Reply:
I tried `LayoutUpdated` at first, but it seemed like events were being received by the handler even before I've even opened the CmdPal/ATS for the first time. I think it's because `LayoutUpdated` fires if anything in the visual tree causes the control to change their "layout"? So I tried `SizeChanged` like the docs said to if I wanted to listen to something more specific to the control and it seems to work.